api_app/core/serializers.py

```diff
@@ -35,63 +21,155 @@ class _ConfigSerializer(rfs.Serializer):
+            .prefetch_related(self.child.Meta.model.snake_case_name)
+            .annotate(
+                value_owner=subquery_owner,
+                value_default=subquery_default,
+                value_organization=subquery_org,
+            )
+        )
+        parsed = defaultdict(list)
+        # populate the result for every plugin (even the ones without parameters)
+        # parsed[plugin]= [parameter1]
+        for parameter in params:
+            parsed[getattr(parameter, self.child.Meta.model.snake_case_name)].append(
+                parameter
+            )
+
+        result = []
+        # we can finally construct our result
+        for plugin in plugins:
+            plugin_representation = self.child.to_representation(plugin)
+            plugin_representation["params"] = {}
+            plugin_representation["secrets"] = {}
+            total_parameter = len(parsed[plugin])
+            parameter_required_not_configured = []
+            for param in parsed[plugin]:
+                # the priority order is
+                # 1 owner
+                # 2 organization
+                # 3 default
+                value = (
+                    param.value_owner or param.value_organization or param.value_default
+                )
+                if param.required and not bool(value):
+                    parameter_required_not_configured.append(param.name)
+                param_representation = ParameterSerializer(param).data
+                if param.is_secret and value == param.value_organization:
+                    value = "redacted"
+                param_representation["value"] = value
+                param_representation.pop("name")
+                if param.is_secret:
+                    plugin_representation["secrets"][param.name] = param_representation
+                else:
+                    plugin_representation["params"][param.name] = param_representation
+            if not parameter_required_not_configured:
+                configured = True
+                details = "Ready to use!"
+            else:
+                details = (
+                    f"{', '.join(parameter_required_not_configured)} "
+                    "secret"
+                    f"{'' if len(parameter_required_not_configured) == 1 else 's'}"
+                    " not set;"
+                    f" ({total_parameter - len(parameter_required_not_configured)} "
+                    f"of {total_parameter} satisfied)"
+                )
+                configured = False
+            if plugin in enabled_plugins:
+                disabled = False
+            else:
+                disabled = True
+            plugin_representation["disabled"] = disabled
+            plugin_representation["verification"] = {
+                "configured": configured,
+                "details": details,
+                "missing_secrets": parameter_required_not_configured,
+            }
+            result.append(plugin_representation)
+
```

Comment:
log of the number of generated plugin representations per user?

Reply:
It is always the number of configurations present in the db. I just that some of them are disabled